here-naksha-lib-handlers/src/main/java/com/here/naksha/lib/handlers/AbstractEventHandler.java

```diff
@@ -34,12 +35,21 @@ public abstract class AbstractEventHandler implements IEventHandler {
 
   protected final @NotNull INaksha nakshaHub;
 
-  public AbstractEventHandler(final @NotNull INaksha hub) {
+  protected AbstractEventHandler(final @NotNull INaksha hub) {
     this.nakshaHub = hub;
   }
 
-  public @NotNull INaksha nakshaHub() {
-    return this.nakshaHub;
+  protected abstract EventProcessingStrategy processingStrategyFor(IEvent event);
+
+  protected abstract @NotNull Result process(@NotNull IEvent event);
+
+  @Override
+  public final @NotNull Result processEvent(@NotNull IEvent event) {
+    return switch (processingStrategyFor(event)) {
+      case PROCESS -> process(event);
+      case SEND_UPSTREAM_WITHOUT_PROCESSING -> event.sendUpstream();
+      case SUCCEED_WITHOUT_PROCESSING -> new SuccessResult();
+    };
```

Comment:
I think we need one more strategy - to allow returning NOT_IMPLEMENTED error (basically simply calling notImplemented() method)

Reply:
Good idea, done :)